### DISCUSSION TITLE
Feature request: Modern Image Format Support

### DISCUSSION THREAD

**Discussion by naanlizard** [OvenMediaLabs/OvenMediaEngine#1738] [Ideas]
Currently, AFAIK OME only supports jpeg and PNG codecs for Image output. 

I would like to request/suggest [WebP](https://caniuse.com/webp) and [AVIF](https://caniuse.com/avif) support as output codecs for images generated by OME. They are much more efficient to serve to users, with smaller file sizes and better compression. They are supported by all modern browsers.

I am thinking specifically about the image encoder, e.g. 

```
    <OutputProfile>
       <Encodes>
           <Video>
                <Bypass>true</Bypass>	
  	   </Video>
           <Image>
               <Codec>jpeg</Codec>
               <Width>1280</Width>
               <Height>720</Height>
               <Framerate>1</Framerate>
          </Image>
       </Encodes>
    </OutputProfile>
```

But perhaps there are other images generated by OME that I'm forgetting.

**Comment by Keukhan** [comment]
@naanlizard  
Thank you for waiting. We now support the modern image format, **WebP**. It has not been officially released yet. Please try it out, and let us know if you have any suggestions for improvement.

Manual : https://airensoft.gitbook.io/ovenmediaengine/dev/thumbnail
Commit : https://github.com/AirenSoft/OvenMediaEngine/commit/26dbba7de74f601c5fa998ebd534df8d0a1f1de9

I wish your success in 2025, and also the success of OvenMediaEngine 👍

**Reply by naanlizard** [reply]
Love to see it! I haven't managed to test this yet, has anyone else? 

Is there a new OME release coming soon?

**Reply by bchah** [reply]
WebP working great on my end 👍

**Reply by naanlizard** [reply]
Same here, though we did run into a new and exciting bug with keepalive, which I'll post about later

Is AVIF planned as well, or just webp? It would help us plan our roadmap

Thanks for the great work Airensoft!

**Reply by Keukhan** [reply]
AVIF will be supported as well. The exact date is not decided yet, but it will likely be added along with AV1 codec support. As you know, AV1 support is planned for this year. :)
